operator: panic observed with v1.5.1

After deploying cilium-operator v1.5.1, I observed the following panic:

```level=info msg="Deleted unused allocator master key" id=4281 key=cilium/state/identities/v1/id/4281 subsys=allocator
panic: reflect: call of reflect.Value.Type on zero Value [recovered]
    panic: reflect: call of reflect.Value.Type on zero Value

goroutine 183 [running]:
encoding/json.(*encodeState).marshal.func1(0xc0008b33f8)
    /usr/local/go/src/encoding/json/encode.go:302 +0x9a
panic(0x1aa7680, 0xc000922160)
    /usr/local/go/src/runtime/panic.go:522 +0x1b5
reflect.Value.Type(0x0, 0x0, 0x0, 0xc00083a210, 0xc000808020)
    /usr/local/go/src/reflect/value.go:1813 +0x169
encoding/json.stringEncoder(0xc000192460, 0x0, 0x0, 0x0, 0xc000800100)
    /usr/local/go/src/encoding/json/encode.go:591 +0x5a
encoding/json.mapEncoder.encode(0x1e62ad8, 0xc000192460, 0x1a8ed40, 0xc000a00a58, 0x195, 0x100)
    /usr/local/go/src/encoding/json/encode.go:702 +0x352
encoding/json.structEncoder.encode(0xc000455300, 0x11, 0x21, 0xc000192460, 0x1d27400, 0xc000a009c0, 0x199, 0x100)
    /usr/local/go/src/encoding/json/encode.go:660 +0x304
encoding/json.structEncoder.encode(0xc000382480, 0x6, 0x8, 0xc000192460, 0x1c1e420, 0xc000a009a0, 0x199, 0x700100)
    /usr/local/go/src/encoding/json/encode.go:660 +0x304
encoding/json.ptrEncoder.encode(0xc000922380, 0xc000192460, 0x1d91640, 0xc000a009a0, 0x16, 0x1d90100)
    /usr/local/go/src/encoding/json/encode.go:806 +0xb2
encoding/json.(*encodeState).reflectValue(0xc000192460, 0x1d91640, 0xc000a009a0, 0x16, 0x1b60100)
    /usr/local/go/src/encoding/json/encode.go:334 +0x82
encoding/json.(*encodeState).marshal(0xc000192460, 0x1d91640, 0xc000a009a0, 0xc001790100, 0x0, 0x0)
    /usr/local/go/src/encoding/json/encode.go:306 +0xee
encoding/json.(*Encoder).Encode(0xc0008b34c8, 0x1d91640, 0xc000a009a0, 0x13, 0x1dab898)
    /usr/local/go/src/encoding/json/stream.go:200 +0x8a
github.com/cilium/cilium/vendor/k8s.io/apimachinery/pkg/runtime/serializer/json.(*Serializer).Encode(0xc0004e22c0, 0x20fb780, 0xc000a009a0, 0x20e1080, 0xc000acc0f0, 0x192b277, 0x13)
    /go/src/github.com/cilium/cilium/vendor/k8s.io/apimachinery/pkg/runtime/serializer/json/json.go:246 +0x2e4
github.com/cilium/cilium/vendor/k8s.io/apimachinery/pkg/runtime/serializer/versioning.DirectEncoder.Encode(0x20e5240, 0xc0004df7e0, 0x20e3a80, 0xc0004e22c0, 0x2100d00, 0xc00016dea0, 0x20fb780, 0xc000a009a0, 0x20e1080, 0xc000acc0f0, ...)
    /go/src/github.com/cilium/cilium/vendor/k8s.io/apimachinery/pkg/runtime/serializer/versioning/versioning.go:263 +0x26f
github.com/cilium/cilium/vendor/k8s.io/apimachinery/pkg/runtime.Encode(0x20e5300, 0xc0007e8b40, 0x20fb780, 0xc000a009a0, 0xc000a009a0, 0x1, 0x0, 0x0, 0x0)
    /go/src/github.com/cilium/cilium/vendor/k8s.io/apimachinery/pkg/runtime/codec.go:46 +0x74
github.com/cilium/cilium/vendor/k8s.io/client-go/rest.(*Request).Body(0xc00075db00, 0x1d91640, 0xc000a009a0, 0xc00075db00)
    /go/src/github.com/cilium/cilium/vendor/k8s.io/client-go/rest/request.go:400 +0x238
github.com/cilium/cilium/pkg/k8s/client/clientset/versioned/typed/cilium.io/v2.(*ciliumNetworkPolicies).Update(0xc000922060, 0xc000a009a0, 0x11, 0x2147ca0, 0xc000922060)
    /go/src/github.com/cilium/cilium/pkg/k8s/client/clientset/versioned/typed/cilium.io/v2/ciliumnetworkpolicy.go:128 +0xfc
main.updateCNP(0x210a2c0, 0xc0004312e0, 0xc000a009a0, 0xc00083be60, 0x10000)
    /go/src/github.com/cilium/cilium/operator/k8s_node.go:291 +0xa59
main.runNodeWatcher.func4.3.1()
    /go/src/github.com/cilium/cilium/operator/k8s_node.go:208 +0xb7
main.runNodeWatcher.func4.2(0xc000b560c0)
    /go/src/github.com/cilium/cilium/operator/k8s_node.go:163 +0x3b
created by main.runNodeWatcher.func4
    /go/src/github.com/cilium/cilium/operator/k8s_node.go:161 +0x218
```

--- a/operator/k8s_node.go
+++ b/operator/k8s_node.go
@@ -205,8 +205,9 @@ func runNodeWatcher() error {
 							}
 							if needsUpdate {
 								wg.Add(1)
+								cnpCpy := cnp.DeepCopy()
 								removeNodeFromCNP <- func() {
-									updateCNP(ciliumK8sClient.CiliumV2(), &cnp, nodesToDelete, k8sCapabilities)
+									updateCNP(ciliumK8sClient.CiliumV2(), cnpCpy, nodesToDelete, k8sCapabilities)
 									wg.Done()
 								}
 							}

